applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_axisymmetric_FIC_element.cpp

```diff
@@ -41,17 +42,8 @@ void UPwSmallStrainAxisymmetricFICElement<TDim, TNumNodes>::CalculateBMatrix(Mat
 {
     KRATOS_TRY
 
-    const double radius = GeoElementUtilities::CalculateRadius(Np, this->GetGeometry());
-
-    for (IndexType i = 0; i < TNumNodes; ++i) {
-        const IndexType index = TDim * i;
-
-        rB(INDEX_2D_PLANE_STRAIN_XX, index + INDEX_X) = GradNpT(i, INDEX_X);
-        rB(INDEX_2D_PLANE_STRAIN_YY, index + INDEX_Y) = GradNpT(i, INDEX_Y);
-        rB(INDEX_2D_PLANE_STRAIN_ZZ, index + INDEX_X) = Np[i] / radius;
-        rB(INDEX_2D_PLANE_STRAIN_XY, index + INDEX_X) = GradNpT(i, INDEX_Y);
-        rB(INDEX_2D_PLANE_STRAIN_XY, index + INDEX_Y) = GradNpT(i, INDEX_X);
-    }
+    AxisymmetricStressState stress_state;
```

Comment:
Yes, this is exactly the way we want to move forward (see also the [discussion PR](https://github.com/KratosMultiphysics/Kratos/pull/11966) that came out of the investigation). We decided it's probably better to first get this small change in and if the other two stress state flavors are in, we can just call it in the base class once and not have all these duplications.

Reply:
(And it's easier to do this when all of the stress states are in, because then we just have an interface in the base class for which we switch the concrete type depending on the element)

Reply:
Thank you for the info.